web/react/components/user_settings/user_settings_general.jsx

```diff
@@ -188,7 +205,8 @@ export default class UserSettingsGeneralTab extends React.Component {
         }
     }
     updateSection(section) {
-        this.setState(assign({}, this.setupInitialState(this.props), {clientError: '', serverError: '', emailError: ''}));
+        const emailChangeInProgress = this.state.emailChangeInProgress;
```

Comment:
Why do we get the current state here then set the current state to what we just got?